runtime/src/accounts_index.rs

```diff
@@ -181,9 +192,20 @@ impl<'a, T: 'static + Clone> Iterator for AccountsIndexIterator<'a, T> {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct MintIndex {
+    index: DashMap<Pubkey, MintIndexEntry>,
+    // Map from pubkey to the mint it's stored in, used for cleanup.
+    // Alternative is to store Option<Pubkey> in each AccountInfo in the
+    // AccountsIndex if something is an SPL account with a mint, but then
+    // every AccountInfo would have to allocate `Option<Pubkey>`
+    accounts_to_mints: DashMap<Pubkey, DashSet<Pubkey>>,
+}
```

Comment:
Given, 20M~50M spl-token accounts, this new secondary index could consume 2G~5G RAM considering 100 bytes per entry (=account) as a rough estimate with safe buffer?